src/peer-routing.js

```diff
@@ -1,40 +1,126 @@
+      this._findClosestPeersTask, this._refreshManagerOptions.interval, this._refreshManagerOptions.bootDelay
+    )
+  }
 
-module.exports = (node) => {
-  const routers = node._modules.peerRouting || []
+  /**
+   * Recurrent task to find closest peers and add their addresses to the Address Book.
+   */
+  async _findClosestPeersTask () {
+    try {
+      for await (const { id, multiaddrs } of this.getClosestPeers(this._peerId.id)) {
+        this._peerStore.addressBook.add(id, multiaddrs)
+      }
+    } catch (err) {
+      log.error(err)
+    }
+  }
 
-  // If we have the dht, make it first
-  if (node._dht) {
-    routers.unshift(node._dht)
+  /**
+   * Stop peer routing service.
+   */
+  stop () {
+    clearDelayedInterval(this._timeoutId)
   }
 
-  return {
-    /**
-     * Iterates over all peer routers in series to find the given peer.
-     *
-     * @param {string} id - The id of the peer to find
-     * @param {object} [options]
-     * @param {number} [options.timeout] - How long the query should run
-     * @returns {Promise<{ id: PeerId, multiaddrs: Multiaddr[] }>}
-     */
-    findPeer: async (id, options) => { // eslint-disable-line require-await
-      if (!routers.length) {
-        throw errCode(new Error('No peer routers available'), 'NO_ROUTERS_AVAILABLE')
+  /**
+   * Iterates over all peer routers in series to find the given peer.
+   *
+   * @param {string} id - The id of the peer to find
+   * @param {object} [options]
+   * @param {number} [options.timeout] - How long the query should run
+   * @returns {Promise<{ id: PeerId, multiaddrs: Multiaddr[] }>}
+   */
+  async findPeer (id, options) { // eslint-disable-line require-await
+    if (!this._routers.length) {
+      throw errCode(new Error('No peer routers available'), 'NO_ROUTERS_AVAILABLE')
+    }
+
+    return pAny(this._routers.map(async (router) => {
+      const result = await router.findPeer(id, options)
+
+      // If we don't have a result, we need to provide an error to keep trying
+      if (!result || Object.keys(result).length === 0) {
+        throw errCode(new Error('not found'), 'NOT_FOUND')
       }
 
-      return pAny(routers.map(async (router) => {
-        const result = await router.findPeer(id, options)
+      return result
+    }))
+  }
+
+  /**
+   * Attempt to find the closest peers on the network to the given key.
+   *
+   * @param {Uint8Array} key - A CID like key
+   * @param {Object} [options]
+   * @param {number} [options.timeout=30e3] - How long the query can take.
+   * @returns {AsyncIterable<{ id: PeerId, multiaddrs: Multiaddr[] }>}
+   */
+  async * getClosestPeers (key, options = { timeout: 30e3 }) {
+    if (!this._routers.length) {
+      throw errCode(new Error('No peer routers available'), 'NO_ROUTERS_AVAILABLE')
+    }
 
-        // If we don't have a result, we need to provide an error to keep trying
-        if (!result || Object.keys(result).length === 0) {
+    const result = await pAny(
```

Comment:
Not going to block this PR on this (as we're doing it in other content/peer routing calls), but we should think about the parallel actions this is doing. Most people shouldn't be using two, but in the change they are, the parallel calls could cause some odd behavior. This is probably worth discussing in more depth when we look at the discovery api / configuration updates.

Reply:
Makes sense, I will cross post it here for us to discuss